improve compare-revs

kmamal opened the pull request.

ronag commented on package.json:
```diff
@@ -95,6 +95,7 @@
     "eslint-plugin-promise": "^6.0.0",
     "eslint-plugin-standard": "^5.0.0",
     "husky": "^8.0.1",
+    "jest": "^28.1.3",
```
why?

kmamal replied:
it's used by the test script

ronag reviewed (changes requested):
Missing benchmark

ronag commented:
I tried simplifying a bit. Please review that and check whether there is performance regression.

ronag commented on util/compare-rev.js:
```diff
@@ -1,29 +1,77 @@
+    }
+
+    result ||= ac === bc ? 0 : ac < bc ? -1 : 1
   }
```
You can skip the dash index

ronag commented on test/util/compare-rev.js:
```diff
@@ -0,0 +1,31 @@
+  ['01-00000000000000', '02-00000000000000', -1],
+  ['02-00000000000000', '01-00000000000000', 1],
+  ['11-00000000000000', '2-00000000000000', 1],
+  ['2-00000000000000', '11-00000000000000', -1],
```
We need test for varying revision lengths. Also you can use the old `compareRev` function to check against. So that we can guarantee the old and new implementations behave the same. Maybe even add a fuzzing test?

kmamal commented:
I hate to say this, but refactoring the common parts of the code into an "impl" function prevents v8 from optimizing it. Probably because it no longer knows the types of it's arguments. Not sure if there's a way around this.

Also very costly are any type of conversion between strings and buffers.

I finally had enough of cronometro and added some diy benchmarking logic.

Also added fuzzing tests.

ronag commented:
Good work!
